### DISCUSSION TITLE
Integration with Radix UI Radio group stopped from version 17.12.2 to  7.13.0

### DISCUSSION THREAD

**Discussion by nigellima** [react-hook-form/react-hook-form#10246] [Q&A] [answered]
### Description
I'm trying to create a Radiogroup component that integrates with react-hook-form. However the only value it's workings is the first one. All the other ones when clickng on them the values are NULL.

This used to work up to the version 17.12.2, but it seems that from 7.13.0 on the issue appeared.

Reproduction:

This is the reproduction working on 17.12.2. Change to version 7.13.0 and the issue will appear
https://codesandbox.io/s/radixui-radio-group-forked-1wlg3x?file=/src/Form.tsx

Thanks in advance

**Comment by fahadsohail482** [comment, marked as answer]
Hey @nigellima You need to use Controller here when integrating custom components with react-hook-form.
With controller we can easily set value to onChange to Form state and this is consistent across all the versions.
I am using recommended way to handle this radio group.
![image](https://user-images.githubusercontent.com/46647496/230711986-015efa02-d983-4b1c-afa5-ac9db81bb63e.png)

There was one thing worth mentioning that the library you are using does have onChange method on the Root component. It only has onValueChange method to access the value.
Here is the fixed version of your code sandbox link [here](https://codesandbox.io/s/radixui-radio-group-forked-ljsg12?file=/src/Form.tsx:867-880)

![image](https://user-images.githubusercontent.com/46647496/230711966-603f7406-6a1d-4e40-b3f3-b5ce29fc833d.png)

Hope you find it useful.
